presto-hive/src/main/java/io/prestosql/plugin/hive/DynamicConfigurationProvider.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive;
+
+import org.apache.hadoop.conf.Configuration;
+
+import java.net.URI;
+
+import static io.prestosql.plugin.hive.HdfsEnvironment.HdfsContext;
+
+public interface DynamicConfigurationProvider
+{
+    Configuration updateConfiguration(Configuration configuration, HdfsContext context, URI uri);
```

Comment:
Why `HdfsContext context, URI uri` parameters are given here? Shouldn't this be implementation specific to each implementation of this interface?

Reply:
This class is used to provide configuration that is specific to the target URI and context. For example, the GCS one will only run if the URI matches "gs://" and will add properties based on the context.

Reply:
But hive does not only talk to HDFS-like components. What about updating configuration to talk to other services like metastore or access control authority?

Maybe I have a problem with this because name `DynamicConfigurationProvider` suggests it is generic, while it is not and it is only used for communication with storage service. Maybe `StorageConfigurationUpdater`.

Reply:
I see the confusion. Perhaps if you think about this as being for a Hadoop `Configuration` rather than generic "configuration". I'm fine with either name.

Today, `HdfsConfigurationUpdater` does a lot more than just "HDFS": it configures compression, text max line length, etc. Hopefully we won't add non-storage things to the new one, but if we do, the more generic name would be better.